x-pack/plugins/security_solution_serverless/server/task_manager/usage_reporting_task.ts

```diff
@@ -109,29 +104,35 @@ export class SecurityUsageReportingTask {
   ) => {
     // if task was not `.start()`'d yet, then exit
     if (!this.wasStarted) {
-      this.logger.debug('[runTask()] Aborted. Task not started yet');
       return;
     }
-
     // Check that this task is current
-    if (taskInstance.id !== this.getTaskId(taskInstance.taskType)) {
+    if (taskInstance.id !== this.taskId) {
       // old task, die
       throwUnrecoverableError(new Error('Outdated task version'));
     }
 
     const [{ elasticsearch }] = await core.getStartServices();
-
     const esClient = elasticsearch.client.asInternalUser;
+
     const lastSuccessfulReport = taskInstance.state.lastSuccessfulReport;
 
-    const usageRecords = meteringCallback({ esClient, lastSuccessfulReport });
+    const usageRecords = await meteringCallback({
+      esClient,
+      cloudSetup: this.cloudSetup,
+      logger: this.logger,
+      taskId: this.taskId,
+      lastSuccessfulReport,
+    });
+    this.logger.info(`received usage records: ${JSON.stringify(usageRecords)}`);
 
     let usageReportResponse: Response | undefined;
 
     try {
       usageReportResponse = await usageReportingService.reportUsage(usageRecords);
-    } catch (e) {
-      this.logger.warn(JSON.stringify(e));
+      this.logger.info(`usage records report was sent successfully`);
```

Comment:
I'm thinking it might make sense to log the response and some of the usage record data on success for compliance and billing debugging.

Reply:
it's good idea to add the response, regarding the usage record data, we already have it in line 127